File: VentasVB/cuProveedor.vb
Public Class cuProveedor
    Dim con As Conexion = New Conexion()
    Private Sub cuProveedor_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        dgvDatos.DataSource = con.datosProveedores()
    End Sub

    Private Sub btnCrear_Click(sender As Object, e As EventArgs) Handles btnCrear.Click
        con.crearProveedor(tbNombre.Text, tbCorreo.Text, tbTelefono.Text, tbCuenta.Text, idUsuario)
        dgvDatos.DataSource = con.datosProveedores()
    End Sub

    Private Sub btnActualizar_Click(sender As Object, e As EventArgs) Handles btnActualizar.Click
        con.actualizarProveedor(Convert.ToInt32(dgvDatos.SelectedRows(0).Cells(0).Value), tbNombre.Text, tbCorreo.Text, tbTelefono.Text, tbCuenta.Text, idUsuario)
        dgvDatos.DataSource = con.datosProveedores()
    End Sub

    Private Sub btnEliminar_Click(sender As Object, e As EventArgs) Handles btnEliminar.Click
        con.eliminarTabla("Proveedor", Convert.ToInt32(dgvDatos.SelectedRows(0).Cells(0).Value), idUsuario)
        dgvDatos.DataSource = con.datosProveedores()
    End Sub

    Private Sub dgvDatos_CellClick(sender As Object, e As DataGridViewCellEventArgs) Handles dgvDatos.CellClick
        tbNombre.Text = dgvDatos.SelectedRows(0).Cells(1).Value.ToString()
        tbCorreo.Text = dgvDatos.SelectedRows(0).Cells(2).Value.ToString()
        tbTelefono.Text = dgvDatos.SelectedRows(0).Cells(3).Value.ToString()
        tbCuenta.Text = dgvDatos.SelectedRows(0).Cells(4).Value.ToString()
    End Sub
End Class

File: Ventas/cuProveedor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Ventas
{
    public partial class cuProveedor : UserControl
    {
        Conexion con = new Conexion();
        public cuProveedor()
        {
            InitializeComponent();
            dgvDatos.DataSource = con.datosProveedores();
        }

        private void btnCrear_Click(object sender, EventArgs e)
        {
            con.crearProveedor(tbNombre.Text, tbCorreo.Text, tbTelefono.Text, tbCuenta.Text, global.id);
            dgvDatos.DataSource = con.datosProveedores();
        }

        private void btnActualizar_Click(object sender, EventArgs e)
        {
            con.actualizarProveedor(Convert.ToInt32(dgvDatos.SelectedRows[0].Cells[0].Value), tbNombre.Text, tbCorreo.Text, tbTelefono.Text, tbCuenta.Text, global.id);
            dgvDatos.DataSource = con.datosProveedores();
        }

        private void dgvDatos_CellClick(object sender, DataGridViewCellEventArgs e)
        {
            tbNombre.Text = dgvDatos.SelectedRows[0].Cells[1].Value.ToString();
            tbCorreo.Text = dgvDatos.SelectedRows[0].Cells[2].Value.ToString();
            tbTelefono.Text = dgvDatos.SelectedRows[0].Cells[3].Value.ToString();
            tbCuenta.Text = dgvDatos.SelectedRows[0].Cells[4].Value.ToString();
        }

        private void btnEliminar_Click(object sender, EventArgs e)
        {
            con.eliminarTabla("Proveedor",Convert.ToInt32(dgvDatos.SelectedRows[0].Cells[0].Value), global.id);
            dgvDatos.DataSource = con.datosProveedores();
        }
    }
}
